chef/cookbooks/keystone/libraries/helpers.rb

```diff
@@ -3,8 +3,11 @@ def self.service_URL(node, host, port)
     "#{node[:keystone][:api][:protocol]}://#{host}:#{port}"
   end
 
-  def self.versioned_service_URL(node, host, port)
-    service_URL(node, host, port) + '/' + node[:keystone][:api][:version] + '/'
+  def self.versioned_service_URL(node, host, port, version=nil)
```

Comment:
I am using this in the nova_dashboard (horizon) barclamp, in an (yet) uncommitted code.